### ISSUE TITLE
Go to Definition now only works if open the entire folder 

### ISSUE THREAD

**Issue by atan4583** [microsoft/pylance-release#1301]
## Environment data

-   VS Code version: 1.52.1
-   Extension version (available under the Extensions sidebar): python-2021.2.633441544
-   OS and version: MacOS Catalina 10.15.7
-   Python version (& distribution if applicable, e.g. Anaconda): 2020.07
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Relevant/affected Python packages and their versions: 3.6
-   Relevant/affected Python-related VS Code extensions and their versions: v2021.2.633441544
-   Value of the `python.languageServer` setting: pylance

## Expected behaviour

Open a single .py file in VS Code **_without_** opening its containing folder, click on Go To Definition should trigger a lookup for the source file in the containing folder of the single .py file

## Actual behaviour

After python extension auto upgraded itself to this v2021.2.633441544, the **_expected behavior stops working_**, now Go To Definition only works if open the entire containing folder of the file in VS code first.   

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. in VS Code with python extension version < v2021.2.633441544, open a single .py file without opening its containing folder, click on "Go To Definition" in the file, the definition source file is brought up from within the same folder where the single .py file resides 

## Logs

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
User belongs to experiment group 'pythonaa'
User belongs to experiment group 'pythonJoinMailingListVar2'
User belongs to experiment group 'pythonJediLSP'
User belongs to experiment group 'pythonDiscoveryModuleWithoutWatcher'
User belongs to experiment group 'pythonTensorboardExperiment'
User belongs to experiment group 'PythonPyTorchProfiler'
User belongs to experiment group 'CollectLSRequestTiming - control'
User belongs to experiment group 'CollectNodeLSRequestTiming - experiment'
User belongs to experiment group 'DeprecatePythonPath - experiment'
Error 2021-05-15 13:51:47: Failed to resolve symbolic link:  [Error: ENOENT: no such file or directory, lstat 'python3.6'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: 'python3.6'
}
Error 2021-05-15 13:51:47: Failed to resolve symbolic link:  [Error: ENOENT: no such file or directory, lstat 'python3.6'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: 'python3.6'
}
> python ~/.vscode/extensions/ms-python.python-2021.2.633441544/pythonFiles/pyvsc-run-isolated.py -c "import sys;print(sys.executable)"
> python ~/.vscode/extensions/ms-python.python-2021.2.633441544/pythonFiles/pyvsc-run-isolated.py -c "import sys;print(sys.executable)"
Python interpreter path: ~/anaconda3/envs/azure/bin/python
Starting Pylance language server.
Error 2021-05-15 14:11:52: Failed to resolve symbolic link:  [Error: ENOENT: no such file or directory, lstat 'python3.6'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: 'python3.6'
}
Error 2021-05-15 14:11:52: Failed to resolve symbolic link:  [Error: ENOENT: no such file or directory, lstat 'python3.6'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: 'python3.6'
}
```

</p>
</details>

**Comment by atan4583** [comment]
Found the **culprit**: **Azure SQLServer extension**.  Indeed the extension causes quite a few serious problems:

- `Go To Definition` stops working if only open a single file without opening the entire folder 
- an annoying lightbulb shows up everywhere in the code, 
- VS Code throws "**couldn't start client python tools**" error every time a .py file is opened in VS Code
- default language server `Jedi` stops working, opening a .py file causes CPU to spike up 100% with fan blasting at full speed, the only way to fix this is to replace jedi with `pylance`

To fix all the above problems, I uninstalled and deleted both **Azure SQLServer extension** and **python extension**  folders from the ~/.vscode/extension folder, then install **python extension**  again.

**Comment by jakebailey** [comment]
I believe that this was fixed in 2021.8.0.

![go to def single file](https://user-images.githubusercontent.com/5341706/134574870-3dc05077-238a-4259-b21f-9b26e2cc8adb.gif)
